src/backend/termion.rs

```diff
@@ -338,3 +416,231 @@ impl fmt::Display for ModifierDiff {
+        assert_eq!(Color::from(tcolor::Rgb(1, 2, 3)), Color::Rgb(1, 2, 3));
+    }
+
+    #[test]
+    fn from_termion_bg() {
+        use tcolor::Bg;
+
+        assert_eq!(
+            Style::from(Bg(tcolor::Reset)),
+            Style::default().bg(Color::Reset)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Black)),
+            Style::default().bg(Color::Black)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Red)),
+            Style::default().bg(Color::Red)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Green)),
+            Style::default().bg(Color::Green)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Yellow)),
+            Style::default().bg(Color::Yellow)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Blue)),
+            Style::default().bg(Color::Blue)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Magenta)),
+            Style::default().bg(Color::Magenta)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Cyan)),
+            Style::default().bg(Color::Cyan)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::White)),
+            Style::default().bg(Color::Gray)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightBlack)),
+            Style::default().bg(Color::DarkGray)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightRed)),
+            Style::default().bg(Color::LightRed)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightGreen)),
+            Style::default().bg(Color::LightGreen)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightBlue)),
+            Style::default().bg(Color::LightBlue)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightYellow)),
+            Style::default().bg(Color::LightYellow)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightMagenta)),
+            Style::default().bg(Color::LightMagenta)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightCyan)),
+            Style::default().bg(Color::LightCyan)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::LightWhite)),
+            Style::default().bg(Color::White)
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::AnsiValue(31))),
+            Style::default().bg(Color::Indexed(31))
+        );
+        assert_eq!(
+            Style::from(Bg(tcolor::Rgb(1, 2, 3))),
+            Style::default().bg(Color::Rgb(1, 2, 3))
+        );
+    }
+
+    #[test]
+    fn from_termion_fg() {
+        use tcolor::Fg;
+
+        assert_eq!(
+            Style::from(Fg(tcolor::Reset)),
+            Style::default().fg(Color::Reset)
+        );
+        assert_eq!(
+            Style::from(Fg(tcolor::Black)),
+            Style::default().fg(Color::Black)
+        );
```

Comment:
We can potentially have a loop with a tuple values for reducing the code repetition but I guess then it is harder to track which test has failed from the output. Either way, I just wanted to ask your opinion on this.

Reply:
I thought of doing that. 

> harder to track which test has failed

True, but for ~100 lines of basically enum testing I would have done it.

Concerning the loop, termion colors are not an enum, each color is a different type and they all implement the `Color` trait.
So I could have tuples of `(dyn Color, _)` or similar. But the `From` trait is implemented on each color struct, not on the `Color` trait.

That's the reason for the macro, will be the same for termwiz and I must say this is annoying.

Reply:
> So I could have tuples of (dyn Color, _) or similar. But the From trait is implemented on each color struct, not on the Color trait.

So you have to use dyn Color then you need to convert it back to a struct, yeah, this is annoying.

> Alternative is to test just a few combinations 

Or we can just keep it like this 😅

Reply:
> convert it back to a struct, yeah, this is annoying

I don't think I could even do it in a loop, because they're all different struct, and downcast take a template parameter ...

Anyway I agree on this :smile:

> Or we can just keep it like this

Reply:
Perhaps consider whether each assert can be a single line by using shorter import aliases (e.g TC::) and stylize shortcuts (e.g. Style::new().on_red())

Reply:
Good idea! It worked, made the tests a lot cleaner and easier to read

Reply:
Or perhaps use .into() to avoid Style::From()

Reply:
This won't work, rustc is unable to infer the target type and requires an annotation.